components/engine/src/iterable.rs

```diff
@@ -12,6 +13,52 @@ pub trait IterOptionsExt {
     fn build_read_opts(self) -> ReadOptions;
 }
 
+struct TsFilter {
+    hint_min_ts: Option<u64>,
+    hint_max_ts: Option<u64>,
+}
+
+impl TsFilter {
+    fn new(hint_min_ts: Option<u64>, hint_max_ts: Option<u64>) -> TsFilter {
+        TsFilter {
+            hint_min_ts,
+            hint_max_ts,
+        }
+    }
+}
+
+impl TableFilter for TsFilter {
+    fn table_filter(&self, props: &TableProperties) -> bool {
+        if self.hint_max_ts.is_none() && self.hint_min_ts.is_none() {
+            return true;
+        }
+
+        let user_props = props.user_collected_properties();
+
+        if let Some(hint_min_ts) = self.hint_min_ts {
+            if let Some(mut p) = user_props.get("tikv.max_ts") {
```

Comment:
better to use `MvccProperties::decode`

Reply:
It will make cycle dependency, I think we should move MvccProperties into some component like `engine_trait` instead of `tikv/src/raftstore/coprocessor/` finally.

Reply:
please add a TODO here

Reply:
add in https://github.com/tikv/tikv/pull/6194/commits/334ef77966589669dbaa32397fbeac08d01e6451